File: algorithms/1572.matrix-diagonal-sum/solution.go
func diagonalSum(mat [][]int) int {
	sum := 0
	l := len(mat)
	i := l / 2

	if l%2 == 1 {
		sum += mat[i][i]
	}

	i -= 1

	for i >= 0 {
		sum += mat[i][i] + mat[i][l-i-1] + mat[l-i-1][i] + mat[l-i-1][l-i-1]
		i -= 1
	}

	return sum
}

File: algorithms/1572.matrix-diagonal-sum/solution.ts
function diagonalSum(mat: number[][]): number {
  const len = mat.length;
  let i = Math.floor(len / 2);
  let sum = 0;

  if (len % 2 == 1) {
    sum += mat[i][i];
  }

  while (i > 0) {
    i--;
    const j = len - i - 1;
    sum += mat[i][i] + mat[i][j] + mat[j][i] + mat[j][j];
  }

  return sum;
}
